chained/dialer.go

```diff
@@ -180,6 +177,12 @@ func (p *proxy) MarkFailure() {
 // involves sending either a CONNECT request or a GET request to initiate a
 // persistent connection to the upstream proxy.
 func defaultDialOrigin(op *ops.Op, ctx context.Context, p *proxy, network, addr string) (net.Conn, error) {
+	select {
+	case <-ctx.Done():
+		log.Debug("Context done for dialing proxy address, returning error.")
+		return nil, ctx.Err()
+	default:
+	}
```

Comment:
This shouldn't be necessary.  Callers should be able to distinguish this with `errors.Is(err, context.DeadlineExceeded)`.  I guess this doesn't hurt though.

Reply:
If the goal is just to log here when that happens, IMO you would be better off (1) always logging the deadline, regardless of whether it has passed and (2) checking the error returned by `reportedDial` below to see if the dial timed out.  This would allow you to also see cases in which the dial was not given sufficient time to complete.

Reply:
OK I'll just kill it for now...theoretically with these changes we shouldn't see those weird negative timeouts...